tests/platform_helper/domain/test_codebase.py

```diff
@@ -0,0 +1,623 @@
+
+    subprocess.run(["git", "init"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    subprocess.run(
+        ["git", "remote", "add", "origin", "git@github.com:uktrade/test-app.git"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+
+    codebase.prepare()
+
+    expected_files_dir = Path(EXPECTED_FILES_DIR) / ".copilot"
+    copilot_dir = Path.cwd() / ".copilot"
+
+    compare_directories = filecmp.dircmp(str(expected_files_dir), str(copilot_dir))
+
+    mocks.echo_fn.assert_has_calls(
+        [
+            call(
+                "File .copilot/image_build_run.sh created",
+            ),
+            call(
+                "File .copilot/config.yml created",
+            ),
+            call(
+                "File phases/build.sh created",
+            ),
+            call(
+                "File phases/install.sh created",
+            ),
+            call(
+                "File phases/post_build.sh created",
+            ),
+            call(
+                "File phases/pre_build.sh created",
+            ),
+        ]
+    )
+
+    assert is_same_files(compare_directories) is True
+
+
+def test_codebase_prepare_does_not_generate_files_in_a_repo_with_a_copilot_directory(tmp_path):
+    mocks = CodebaseMocks()
+    mocks.load_application_fn.side_effect = SystemExit(1)
+    codebase = Codebase(**mocks.params())
+    os.chdir(tmp_path)
+    Path(tmp_path / "copilot").mkdir()
+
+    subprocess.run(["git", "init"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    subprocess.run(
+        ["git", "remote", "add", "origin", "git@github.com:uktrade/some-test-app.git"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+
+    codebase.prepare()
+
+    mocks.echo_fn.assert_has_calls(
+        [
+            call(
+                "You are in the deploy repository; make sure you are in the application codebase repository.",
+            ),
+        ]
+    )
+
+
+def test_codebase_build_does_not_trigger_build_without_an_application():
+    mocks = CodebaseMocks()
+    mocks.load_application_fn.side_effect = ApplicationNotFoundError()
+    codebase = Codebase(**mocks.params())
+
+    with pytest.raises(click.Abort) as exc:
+        codebase.build("not-an-application", "application", "ab1c23d")
+        mocks.echo_fn.assert_has_calls(
+            [
+                call(
+                    """The account "foo" does not contain the application "not-an-application"; ensure you have set the environment variable "AWS_PROFILE" correctly.""",
+                    fg="red",
+                ),
+            ]
+        )
+
+
+def test_codebase_prepare_does_not_generate_files_in_a_repo_with_a_copilot_directory(tmp_path):
+    mocks = CodebaseMocks()
+    mocks.load_application_fn.side_effect = SystemExit(1)
+    codebase = Codebase(**mocks.params())
+    os.chdir(tmp_path)
+    Path(tmp_path / "copilot").mkdir()
+
+    subprocess.run(["git", "init"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
like we inject `load_application_fn` could we inject subprocess to mock? Maybe I am not clear on how you are currently testing with subprocess?